projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Inline Help.
+ *
+ * Handles providing a LiveChat icon within WPAdmin until such time
+ * as the full live chat experience can be run in a non-Calypso environment.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+/**
+ * Class Inline_Help.
+ */
+class Inline_Help {
+
+	/**
+	 * Inline_Help constructor.
+	 */
+	public function __construct() {
+		$this->register_actions();
+	}
+
+	/**
+	 * Registers actions.
+	 *
+	 * @return void
+	 */
+	public function register_actions() {
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		// Do not inject the FAB icon on embedded screens since the parent window may already contain a FAB icon.
+		$is_framed = ! empty( $_GET['frame-nonce'] );
+
+		if ( $is_framed ) {
+			return;
+		}
+		// phpcs:enable WordPress.Security.NonceVerification.Recommended
+
+		add_action( 'admin_footer', array( $this, 'add_fab_icon' ) );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'add_fab_styles' ) );
+	}
+
+	/**
+	 * Outputs "FAB" icon markup and SVG.
+	 *
+	 * @return void|string the HTML markup for the FAB or early exit.
+	 */
+	public function add_fab_icon() {
+
+		if ( wp_doing_ajax() ) {
+			return;
+		}
+
+		$svg_allowed = array(
+			'svg'   => array(
+				'id'              => true,
+				'class'           => true,
+				'aria-hidden'     => true,
+				'aria-labelledby' => true,
+				'role'            => true,
+				'xmlns'           => true,
+				'width'           => true,
+				'height'          => true,
+				'viewbox'         => true, // <= Must be lower case!
+			),
+			'g'     => array( 'fill' => true ),
+			'title' => array( 'title' => true ),
+			'path'  => array(
+				'd'    => true,
+				'fill' => true,
+			),
+		);
+
+		$gridicon_help = file_get_contents( __DIR__ . '/gridicon-help.svg', true );
+
+		// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
+		// We trust that output in the template has been escaped.
+		echo load_template(
+			__DIR__ . '/inline-help-template.php',
+			true,
+			array(
+				'href'                => 'https://wordpress.com/help',
+				'tracking_event_name' => 'jetpack_masterbar_inline_help_click',
+				'icon'                => $gridicon_help,
+				'svg_allowed'         => $svg_allowed,
+			)
+		);
```

Comment:
[`load_template` is not very used in Jetpack](https://github.com/Automattic/jetpack/search?q=load_template), and I noted there is a [preference for using `printf`](https://github.com/Automattic/jetpack/search?q=printf) instead. Do you have any strong opinion against it?
```suggestion
		printf(
			'<div class="a8c-faux-inline-help"><a href="%1$s" target="_blank" rel="noopener noreferrer" title="%2$s" class="a8c-faux-inline-help__button jptracks" data-jptracks-name="%3$s">%4$s</a></div>',
			esc_url( 'https://wordpress.com/help' ),
			esc_attr__( 'Help', 'jetpack' ),
			esc_attr( 'jetpack_masterbar_inline_help' ),
			wp_kses( $gridicon_help, $svg_allowed )
		);
```

That should also get rid of the necessity of disabling the `WordPress.Security.EscapeOutput.OutputNotEscaped` rule.

Reply:
Personally, I find it much easier to maintain HTML in a dedicated file and simply drop into PHP when I need to. 

Indeed, that's what WordPress itself does 90% of the time when dealing with markup.

There is a convention in WP land where people like to manipulate strings of markup within PHP. I would personally avoid this at all costs but if it's not the JP way then that's fine. 

> WordPress.Security.EscapeOutput.OutputNotEscaped

This is a tool - sometimes it's "wrong" 😄

Reply:
> I would personally avoid this at all costs but if it's not the JP way then that's fine.

I'm going to defer to @jeherve on this one, since I'm not sure if there is any guideline regarding this (I just observed that preference by looking at the existing code).

Reply:
Happy either way. If it avoids problems I'm fine to covert it back.